### ISSUE TITLE
Make caja remember last session folders

### ISSUE THREAD

**Issue by colibris79** [mate-desktop/caja#523]
Hi
I would like that caja remembered the folders and tabs that were open when I close the window, so when I reopen caja it starts where I was last time. I've been trying to get this but haven't been able to do it so far. Is it possible?
Thanks

**Comment by Rubenzahl** [comment]
I would love to have this too!!!!

**Comment by lukefromdc** [comment]
If you are taking about having Caja remember what folders were open in the previous SESSION, it can do that now as this is a feature GNOME 2 always had. Yes, it still works, even in GTK3 builds. 

To enable it, open dconf-editor, go to org>mate>desktop>session, set auto-save-session to "true" and any folder open when you log out will open again when you log back in. Works with pluma and mate-terminal too.

**Comment by colibris79** [comment]
Hi lukefromdc. Thanks for your answer. It is not exactly what I was looking for but it is still a very cool feature. 
What I would like is something like the browsers options that you close the window and when you open it again you get the same tabs that you had when you closed it. So you remain in the same session, you just close the caja window.

**Comment by skjnldsv** [comment]
Same here!

**Comment by Rubenzahl** [comment]
@colibris79 
This can be done in the browser itself. In FF you go to settings: Section: Start: "Remember last session Tabs" (or similar - have different language version)

**Comment by GenZai** [comment]
That would be awesome

**Comment by larry2311** [comment]
Ubuntu 16.04 version Caja 1.12.7 does not seem to have "auto-save-session" in dconf-editor.
Having the File Manager able to save and restore the previous session, I would have thought to be essential.

However, after many years is still not implemented (as far as I can see), in the Caja File Manager, or any other File Manager for that matter.

What is happening?

**Comment by lukefromdc** [comment]
We have it in MATE 1.16, 1.18, and 1.20, I can pull it up at org>mate>desktop>session and its in that item, not in the "required components" below it

**Comment by Rubenzahl** [comment]
The auto-save feature is hidden here:

In dconf-editor:


For mate:

org -> mate -> desktop -> session

auto-save-session (needs to be ticked)

But from time to time *I had some problems with this*. The session had
not been saved. But now it seems to be working again. I do not know why
it sometimes works and sometimes doesn't.


For cinnamon:

https://forums.linuxmint.com/viewtopic.php?t=225769

Good luck!




Am 14.05.2018 um 10:18 schrieb larry2311:

**Comment by skjnldsv** [comment]
@Rubenzahl no, this option save the entire desktop.
We're talking about a "remember opened tabs" option for caja only. Like chromium or firefox :)

**Comment by L-U-T-i** [comment]
I remember this was working someyears ago (don't remember exact versions though), but caja windows are not saved/restored in most recent versions (1.24 & 1.25) - at least on my systems (CentOS / Fedora). I have "Auto Save Session" enabled in dconf. Terminal windows and Pluma are restored OK (not always for VNC sessions though).

Is there a way to also save/restore caja windows?

**Comment by lukefromdc** [comment]
To save a session you need to properly log out of it. Killing the xserver, a forced poweroff, or anything that terminates the session without logging out of it will cause it not to be saved. Usual effect of this when session saving is enabled is the LAST saved session is what comes up, at least in my tests(not that I don't normally use this but do test it as I was the one that ported the code in caja to GtkApplication.)

**Comment by L-U-T-i** [comment]
Is it expected not to save /restore caja windows for root user session (with "caja --force-desktop") at all?

Or, better said - is there an option to get opened windows saved and restored also for root user (caja --force-desktop)?

If it is like that by design, would it be possible to simply resolve it with a custom patch / build? If yes, which part of the code is to be changed?

**Comment by raveit65** [comment]
> root user session 

crazy and dangerous....

**Comment by L-U-T-i** [comment]
Don't worry, I know what I'm doing... :-)

1. It is a development server, accessible only on internal IP, and
2. I have 2 VNC sessions:
- root for what I would do with "sudo caja&" anyway (just in a more simple way ;-)) and more (some programs behave differently through a VNC session and simply don't open for a regular user - instead of asking for credentials as normally...), and
- normal user for 99,9% of time / tasks

Furthermore, it is very simple to make backups for a virtual machine, for the case of a fatal stupid mistake (which you can easily do also within a normal user session, with "sudo something" or "su root"... ;-)).

Doing this way about 15 years now, and proven to be the most efficient for me...

**Comment by lukefromdc** [comment]
Thers are of course uses for caja as root. One of them is of course
repairing such problems in rolling release distros as updates removing
"obsolete" files needed ro build something, or cleaning up leftovers
from things in /usr/local.
Another oddly enough is packing up my Debian packages from files made
by checkinstall, which can't quite do the job by itself due to picking
up excess files.
Oh yeah I keep a bootable backup partition handy at all times, bad
things happen on development machines.

On 7/16/2020 at 6:44 AM, "raveit65"  wrote:
	root user session  

	crazy and dangerous....

**Comment by raveit65** [comment]
@lukefromdc 
I am talking abiut a `root user sesson

> Thers are of course uses for caja as root. One of them is of course repairing such problems in rolling release distros as updates removing "obsolete" files needed ro build something, or cleaning up leftovers from things in /usr/local. Another oddly enough is packing up my Debian packages from files made by checkinstall, which can't quite do the job by itself due to picking up excess files. Oh yeah I keep a bootable backup partition handy at all times, bad things happen on development machines. On 7/16/2020 at 6:44 AM, "raveit65" wrote: root user session crazy and dangerous.... — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

@lukefromdc 
I meant a `root user mate session` not using caja with `sudo' or `su`.
I hope you don't use MATE with a root session, because you said you're are under us-facist radar.

**Comment by lukefromdc** [comment]
If I needed to deal with a gui root session for any reason (e.g.
testing it...) first order of business would be to ensure no Internet
connection is available no matter what the machine does. With only USB
devices for handling wireless and no landline(oe ethernet setup) this
is easy-pull the plug. Same for anyone w only wired access. 
A laptop with internal wireless on the other hand cannot safely enter
a root session where an  open or previously used connection is
available, unless there is a hardware RF kill on it and you trust that
that actually works all the way.

On 7/16/2020 at 1:58 PM, "raveit65"  wrote:
	@lukefromdc
 I am talking abiut a `root user sesson  

	Thers are of course uses for caja as root. One of them is of course
repairing such problems in rolling release distros as updates removing
"obsolete" files needed ro build something, or cleaning up leftovers
from things in /usr/local. Another oddly enough is packing up my
Debian packages from files made by checkinstall, which can't quite do
the job by itself due to picking up excess files. Oh yeah I keep a
bootable backup partition handy at all times, bad things happen on
development machines. On 7/16/2020 at 6:44 AM, "raveit65" wrote: root
user session crazy and dangerous.... — You are receiving this
because you commented. Reply to this email directly, view it on
GitHub, or unsubscribe.  

	@lukefromdc
 I meant a root user mate session not using caja with sudo' or su`.
 I hope you don't use MATE with a root session, because you said
you're are under us-facist radar.

**Comment by raveit65** [comment]
@lukefromdc 
Please read the whole topic.
I didn't talk about a laptop/box without internet connection.
The question was about a server using a root session with a server with a internet connection.

**Comment by lukefromdc** [comment]
I did read the whole topic, you said root sessions are more dangerous to those facing external threats (government in this case) but I replied that in most cases the box (if this is a LOCAL box anyway...) can be taken offline temporarily to avoid that threat. User error of course can still kill the system.  

People do use root sessions for various purposes, and we are probably in agreement that that should never be done online. I DO remember a long time ago (2004) having a Debian box someone set up for me where the normal way to use Synaptic was from a root session (ugly and dangerous).

As for servers, using a GUI for a remote root session (if this is what you mean) strikes me as especially dangerous. This goes double for a webserver or other publicly known box. If that's what we are being asked to support, it is nuts, and a waste of our effort to support/promote.

**Comment by L-U-T-i** [comment]
I agree GUI root session is not to be used on a production servers, absolutely.

But, as said - for a development server (many root tasks involved among the other...) in a safe local environment, it just simplify things drastically sometimes (can do as much harm also from normal user GUI session with 'su root' / 'sudo ...', if doing stupid things... ).

Considering it is a virtual machine, (local / wired LAN) VNC is for me the only suitable option I've found by now (Spice server also seems to have some locale / key mapping issues...). So, it is just technically a remote session...

At the end, as long as upstream (RedHat) has no problem allowing and even prividing login as root, I really don't see why MATE desktop would be so much more restrictive / take more care for me than the system it is within / a part of.

I also agree this is not to be promoted anyhow. But, such features can be needed sometimes, as many other things in our life (to be able to temporary turn off some safety measures to do some task that would be impossible - or at least much more complicated - otherwise). If you are a clever person, you just don't use such options (despite they are available) unless you have to. If you are not, you learn on a hard way, as generally in your life...

At the end, I love linux (and, hate winblow$ , after NT 4 ages at least) just because of that - that the operating sistem doesn't know better what is good for me than myself... ;-)

**Comment by cdeai** [comment]
I am bringing this old feature request again, since it is a long wished-for feature that is sadly still missing.

The feature I am talking about is **Caja remembering its last opened tabs and restoring them when you open it next time - after restart _or_ after closing it normally _or_ after it crashes**.

This is a feature we are used from virtually all modern tabbed web browsers. 
It is also implemented in some file managers, such as SpaceFM, which is a great file manager, but not as integrated into the MATE desktop as Caja is. Therefore, this functionality would be wonderful to have also in Caja, so I am humbly bringing this topic up again.

**Comment by surfrock66** [comment]
Seconded; if there was a way for caja to periodically tattoo the list of open tabs to the config, then read that on open if the option is enabled, rebooting when I am in the middle of a large project would be much much easier
